### ISSUE TITLE
New check: JavadocRegexp

### ISSUE THREAD

**Issue by gianmarcoschifone** [checkstyle/checkstyle#20987]
# Description
After https://github.com/checkstyle/checkstyle/issues/19913

The Documentation Comments Style Guide recommends avoiding Latin abbreviations in Javadoc comments:

- use "also known as" instead of "aka";
- use "that is" or "to be specific" instead of "i.e.";
- use "for example" instead of "e.g.";
- use "in other words" or "namely" instead of "viz.".

This check should report Javadoc comments whose content matches a configured regular expression. This makes it possible to enforce the "Avoid Latin" rule and other project-specific wording rules in Javadocs.

The check should support these properties:

- `format`: regular expression to match forbidden Javadoc content. This is the main property used to define what text is not allowed.
- `ignoreCase`: controls whether `format` should be matched case-insensitively. For the "Avoid Latin" rule, this allows the same pattern to catch `e.g.`, `E.g.`, `AKA`, and similar case variants.
- `ignoreMarkup`: controls what Javadoc text is matched. When `false`, the check matches raw Javadoc source exactly as it appears in the file, including Javadoc tags, HTML tags, and HTML attributes. This is useful for source-format rules, such as forbidding `<br>` tags. When `true`, the check ignores Javadoc/HTML markup and matches only the documentation text visible to a reader. This is useful for wording rules, such as avoiding Latin abbreviations without reporting matches inside links or markup.

### Default values:
- `format`: `$.`
- `ignoreCase`: `false`
- `ignoreMarkup`: `false`

# Config

Configuration for matching visible Javadoc text:

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocRegexp">
            <property name="format" value="(^|\W)(aka|i\.e\.|e\.g\.|viz\.)(\W|$)"/>
            <property name="ignoreCase" value="true"/>
            <property name="ignoreMarkup" value="true"/>
        </module>
    </module>
</module>
```

Configuration for matching raw Javadoc source:

```xml
<module name="Checker">
    <module name="TreeWalker">
        <module name="JavadocRegexp">
            <property name="format" value="&lt;br\s*/?&gt;"/>
            <property name="ignoreMarkup" value="false"/>
        </module>
    </module>
</module>
```

# Example

Example with `ignoreMarkup` set to `true`:

```java
public class Test {

    /**
     * See <a href="https://example.com/aka">documentation</a>.
     */
    public void valid() { }

    /**
     * Creates a user, aka an account owner. // violation
     */
    public void invalid() { }
}
```

Output:

```
Starting audit...
[ERROR] Test.java:/:/: Javadoc content matches the illegal pattern '(^|\W)(aka|i\.e\.|e\.g\.|viz\.)(\W|$)'. [JavadocRegexp]
Audit done.
Checkstyle ends with 1 errors.
```

Example with `ignoreMarkup` set to `false`:

```java
public class Test {

    /**
     * Uses a paragraph tag.
     *
     * <p>The value is normalized.</p>
     */
    public void valid() { }

    /**
     * Uses an HTML line break. <br> // violation
     */
    public void invalid() { }
}
```

Output:

```
Starting audit...
[ERROR] Test.java:/:/: Javadoc content matches the illegal pattern '<br\s*/?>'. [JavadocRegexp]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by romani** [comment]
consider to do JavadocRegexp to operate over source of javadoc.

"content" looks like a text that is seen in html rendering.

**Comment by romani** [comment]
new Check is merged, 

@gianmarcoschifone , now we unblocked to use it in style guide and update coverage table.
